src/USER-HSMA/hsma2d.h

```diff
@@ -0,0 +1,113 @@
+/* -*- c++ -*- ----------------------------------------------------------
+  LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+  http://lammps.sandia.gov, Sandia National Laboratories
+  Steve Plimpton, sjplimp@sandia.gov
+
+  Copyright (2003) Sandia Corporation.  Under the terms of Contract
+  DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+  certain rights in this software.  This software is distributed under
+  the GNU General Public License.
+
+  See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+/* ----------------------------------------------------------------------
+   Contributing author: Jiuyang Liang (liangjiuyang@sjtu.edu.cn)
+------------------------------------------------------------------------- */
+
+#ifdef KSPACE_CLASS
+
+KSpaceStyle(HSMA2D, HSMA2D)
+
+#else
+
+#ifndef LMP_HSMA2D_H
+#define LMP_HSMA2D_H
+
+#include "kspace.h"
+
+#include <cstdio>
```

Comment:
Please do not add include files beyond the absolutely necessary in the "style header" and use forward declarations as much as possible. Instead include those files and have the full definitions in the corresponding .cpp files. Please see
https://github.com/lammps/lammps/blob/master/doc/include-file-conventions.md

Reply:
Modified based on https://github.com/lammps/lammps/blob/master/doc/include-file-conventions.md.